include/zephyr/drivers/clock_control/clock_control_silabs.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2024 Silicon Laboratories Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_CLOCK_CONTROL_SILABS_H_
+#define ZEPHYR_INCLUDE_DRIVERS_CLOCK_CONTROL_SILABS_H_
+
+#include <zephyr/drivers/clock_control.h>
+
+#if defined(CONFIG_SOC_SERIES_EFR32MG21)
+#include <zephyr/dt-bindings/clock/silabs/xg21-clock.h>
+#elif defined(CONFIG_SOC_SERIES_EFR32BG22)
+#include <zephyr/dt-bindings/clock/silabs/xg22-clock.h>
+#elif defined(CONFIG_SOC_SERIES_EFR32MG24)
+#include <zephyr/dt-bindings/clock/silabs/xg24-clock.h>
+#elif defined(CONFIG_SOC_SERIES_EFR32BG27)
+#include <zephyr/dt-bindings/clock/silabs/xg27-clock.h>
+#endif
```

Comment:
Is this inclusion needed? It seems `clock_control_silabs_series.c` doesn't need it. I feel it is only needed for `tests/`. In this case, maybe we can move these lines.

Reply:
Yeah, it's only strictly needed for the tests. The only reason to have them here would be if anyone ever wants to use clock control directly from C, without going through DT to construct the subsys struct.

Reply:
But I did it for similarity with the other clock_control drivers, who (almost) all include the DT bindings in their headers.

Reply:
With the change to using FIELD_PREP, the mask macros are now shared between DT and the driver, so this header now _does_ need to include the DT bindings.